src/cmd/job/attach.c

```diff
@@ -347,6 +355,55 @@ static void handle_output_log (struct attach_ctx *ctx,
     }
 }
 
+void json_decref_wrapper (void *data)
+{
+    json_t *o = data;
+    json_decref (o);
+}
+
+void store_tail_output (struct attach_ctx *ctx,
+                        json_t *entry,
+                        json_t *context)
+{
+    if (context) {
+        bool eof;
+        if (iodecode (context, NULL, NULL, NULL, NULL, &eof) < 0)
+            log_msg ("malformed event context");
+        if (eof)
+            return;
+    }
+    json_incref (entry);
+    if (zlist_append (ctx->tail_output, entry) < 0)
+        log_err_exit ("zlist_append");
+    zlist_freefn (ctx->tail_output,
+                  entry,
+                  json_decref_wrapper,
```

Comment:
You could drop the `json_decref_wrapper()` and use `(zlist_free_fn *) json_decref` here instead.